src/main/java/ca/uhn/fhir/jpa/starter/service/HelperService.java

```diff
@@ -405,6 +409,89 @@ public ResponseEntity<LinkedHashMap<String, Object>> createUsers(@RequestParam("
 				map.put("CSV length validation failed", s);
 				continue;
 			}
+
+			String givenState = state;
+			String givenLga = lga;
+			String givenWard = ward;
+			GroupRepresentation keycloakGroup;
+
+
+			Map<String, Integer> levelNameOccurrences = new HashMap<>();
+
+			Map<String, String> levels = new LinkedHashMap<>(); // Use a LinkedHashMap
+			levels.put("countryName", countryName);
+			levels.put("givenState", givenState);
+			levels.put("givenLga", givenLga);
+			levels.put("givenWard", givenWard);
+
+			// Count variable occurrences
+			for (String level : levels.values()) {
+				levelNameOccurrences.put(level, levelNameOccurrences.getOrDefault(level, 0) + 1);
+			}
+
+			List<String> repeatedLevelNames = new ArrayList<>();
+
+			for (Map.Entry<String, Integer> entry : levelNameOccurrences.entrySet()) {
+				if (entry.getValue() > 1) {
+					repeatedLevelNames.add(entry.getKey());
+				}
+			}
+
+			if (!repeatedLevelNames.isEmpty()) {
+				List<String> levelsWithSameName = new ArrayList<>();
+
+				for (String equalVariable : repeatedLevelNames) {
+					List<String> matchingKeys = new ArrayList<>();
+
+					for (Map.Entry<String, String> entry : levels.entrySet()) {
+						if (entry.getValue().equals(equalVariable)) {
+							matchingKeys.add(entry.getKey());
+						}
+					}
+					// Remove the first value from the list
+					if (!matchingKeys.isEmpty()) {
+						matchingKeys.remove(0);
+					}
+					levelsWithSameName.addAll(matchingKeys);
+				}
+
+				List<String> keysList = new ArrayList<>(levels.keySet());
+
+				for (String level : levelsWithSameName) {
+					switch (level) {
+						case "givenState" :
+							givenState = countryName + "_" + givenState;
+							keycloakGroup = getKeycloakGroupId(givenState);
+							if(null != keycloakGroup && keycloakGroup.getAttributes().get("type").get(0).equals("state")) {
+								state = givenState;
+							} else {
+								map.put("State is not found", s);
+								continue outer;
+							}
+
+						case "givenLga" :
+							givenLga = givenState + "_" + givenLga;
+							keycloakGroup = getKeycloakGroupId(givenLga);
+							if(null != keycloakGroup && keycloakGroup.getAttributes().get("type").get(0).equals("lga")) {
+								lga = givenLga;
+							} else {
+								map.put("Lga is not found", s);
+								continue outer;
+							}
+
+						case "givenWard" :
+							givenWard = givenLga + "_" + givenWard;
+							keycloakGroup = getKeycloakGroupId(givenWard);
+							if(null != keycloakGroup && keycloakGroup.getAttributes().get("type").get(0).equals("ward")) {
+								ward = givenWard;
+							} else {
+								map.put("Ward is not found", s);
+								continue outer;
+							}
+					}
+				}
+			}
```

Comment:
These changes fix the issue but the code logic is bit complex to understand. Can you check if it can be simplified by extracting this part to a different function. Otherwise use comments to describe the functionality. It will help for future updates.

Reply:
Extracting it to a different function makes the code complex since i am skipping the while loop inside switch case. 
I can add comments